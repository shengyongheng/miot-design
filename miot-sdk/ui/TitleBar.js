import { TitleBar } from 'mhui-rn';
export default TitleBar;