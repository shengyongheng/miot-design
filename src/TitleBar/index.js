// @ts-ignore
import { TitleBar } from 'mhui-rn-fixed';
import React from 'react';

const Index = () => {
  return (
    <TitleBar
      type="light"
      title="title"
      subTitle="subtitle"
      style={{
        height: 65,
        backgroundColor: '#222',
      }}
      onPressLeft={() => {
        // navigation.goBack();
      }}
      onPressLeft2={() => console.log('onPressLeft2')}
      onPressRight={() => {
        // navigation.navigate('moreMenu', { title: '设置' });
      }}
      onPressRight2={() => console.log('onPressRight2')}
      showDot={true}
    />
  );
};

export default Index;
