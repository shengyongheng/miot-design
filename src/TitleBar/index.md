---
title: TitleBar
group:
  path: '/basic'
  title: 基础组件
  order: 2
nav:
  title: '组件'
  path: /components
  order: 0
---

# TitleBar 导航栏

## 简介

米家通用导航栏，目前正在开发新版导航栏

## 用法

```jsx
import React from 'react';
import { Text } from 'react-native';
import TitleBar from 'miot/ui/TitleBar';

export default () => (
  <>
    <TitleBar
      type="light"
      title="title"
      subTitle="subtitle"
      style={{
        height: 45,
        backgroundColor: '#222',
      }}
      onPressLeft={() => {
        // navigation.goBack();
      }}
      onPressLeft2={() => console.log('onPressLeft2')}
      onPressRight={() => {
        // navigation.navigate('moreMenu', { title: '设置' });
      }}
      onPressRight2={() => console.log('onPressRight2')}
      showDot={true}
    />
  </>
);
```

## API

| 属性  | 类型    | 默认值    | 说明                                           |
| ----- | ------- | --------- | ---------------------------------------------- |
| type  | string  | 'default' | 按钮类型，可选值为 'primary', 'dashed', 'link' |
| size  | string  | 'middle'  | 按钮尺寸，可选值为 'large', 'middle', 'small'  |
| shape | string  | 'default' | 按钮形状，可选值为 'circle', 'round'           |
| value | boolean | false     | 指定当前是否选中                               |
| ...   | ...     | ...       | ...                                            |
