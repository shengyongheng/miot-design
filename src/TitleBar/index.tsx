// @ts-ignore
import TitleBar from 'miot/ui/TitleBar';
import React from 'react';

const Index: React.FC<any> = () => {
  return (
    <TitleBar
      type="light"
      title="title"
      subTitle="subtitle"
      style={{
        height: 65,
        backgroundColor: '#222',
      }}
      onPressLeft={() => {
        // navigation.goBack();
      }}
      onPressLeft2={() => console.log('onPressLeft2')}
      onPressRight={() => {
        // navigation.navigate('moreMenu', { title: '设置' });
      }}
      onPressRight2={() => console.log('onPressRight2')}
      showDot={true}
    />
  );
};

export default Index;
