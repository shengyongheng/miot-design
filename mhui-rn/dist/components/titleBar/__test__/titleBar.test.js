
import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TitleBar from '../TitleBar';

configure({ adapter: new Adapter() });

describe('render', () => {
  it('render TitleBar', () => {
    const tree = renderer.create(
      <TitleBar
        type="dark"
        title="title"
        subTitle="subTitle"
        onPressLeft={() => { console.log('onPressLeft'); }}
        onPressLeft2={() => { console.log('onPressLeft2'); }}
        onPressRight={() => { console.log('onPressRight'); }}
        onPressRight2={() => { console.log('onPressRight2'); }}
        onPressTitle={() => { console.log('onPressTitle'); }}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
