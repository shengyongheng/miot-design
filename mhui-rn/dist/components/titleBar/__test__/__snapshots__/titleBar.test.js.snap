// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render render TitleBar 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "white",
      "flexDirection": "row",
      "height": 117,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
      "width": 750,
    }
  }
>
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../src/resources/title/std_tittlebar_main_device_back_normal.png",
      }
    }
    style={
      Array [
        Object {
          "height": 28,
          "marginHorizontal": 14,
          "marginVertical": 13.5,
          "resizeMode": "contain",
          "width": 28,
        },
        Object {
          "height": 28,
        },
      ]
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../src/resources/title/std_tittlebar_main_device_back2_normal.png",
      }
    }
    style={
      Array [
        Object {
          "height": 28,
          "marginHorizontal": 14,
          "marginVertical": 13.5,
          "resizeMode": "contain",
          "width": 28,
        },
        Object {
          "height": 28,
          "marginLeft": 0,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "flex": 1,
          "height": 55,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      numberOfLines={1}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000000cc",
            "fontSize": 15,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
        ]
      }
    >
      title
    </Text>
    <Text
      numberOfLines={1}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#00000088",
            "fontSize": 12,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
        ]
      }
    >
      subTitle
    </Text>
  </View>
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../src/resources/title/std_tittlebar_main_device_share_normal.png",
      }
    }
    style={
      Array [
        Object {
          "height": 28,
          "marginHorizontal": 14,
          "marginVertical": 13.5,
          "resizeMode": "contain",
          "width": 28,
        },
        Object {
          "height": 28,
          "marginRight": 0,
        },
      ]
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../src/resources/title/std_tittlebar_main_device_more_normal.png",
      }
    }
    style={
      Array [
        Object {
          "height": 28,
          "marginHorizontal": 14,
          "marginVertical": 13.5,
          "resizeMode": "contain",
          "width": 28,
        },
        Object {
          "height": 28,
        },
      ]
    }
  />
</View>
`;
